From grundler@lackof.org Sun Oct 08 20:17:07 2006
Received: (at 391384) by bugs.debian.org; 9 Oct 2006 03:17:07 +0000
Return-path: <grundler@lackof.org>
Received: from colo.lackof.org ([198.49.126.79])
	by spohr.debian.org with esmtp (Exim 4.50)
	id 1GWlda-00011G-H1
	for 391384@bugs.debian.org; Sun, 08 Oct 2006 20:17:06 -0700
Received: from localhost (localhost [127.0.0.1])
	by colo.lackof.org (Postfix) with ESMTP id 8584B3D836F;
	Sun,  8 Oct 2006 21:16:35 -0600 (MDT)
Received: from colo.lackof.org ([127.0.0.1])
	by localhost (colo.lackof.org [127.0.0.1]) (amavisd-new, port 10024)
	with ESMTP id 02929-04; Sun, 8 Oct 2006 21:16:35 -0600 (MDT)
Received: by colo.lackof.org (Postfix, from userid 27253)
	id 26D0F3D836C; Sun,  8 Oct 2006 21:16:35 -0600 (MDT)
Date: Sun, 8 Oct 2006 21:16:35 -0600
From: Grant Grundler <grundler@parisc-linux.org>
To: Matt Taggart <taggart@debian.org>
Cc: James Bottomley <James.Bottomley@SteelEye.com>,
	dann frazier <dannf@debian.org>, grundler@parisc-linux.org,
	391384@bugs.debian.org
Subject: Re: Bug#391384: linux-image-2.6.18-1-686: Compaq Proliant DL380 fails to boot
Message-ID: <20061009031635.GA30517@colo.lackof.org>
References: <ddc185be0610060407y58c9e64cvca5c69dde88c3703@mail.gmail.com> <20061006203449.GC13984@colo> <20061008214011.2F96F1AAE02@cyrix.home.bogus> <1160347341.3528.4.camel@mulgrave.il.steeleye.com> <20061008231659.19EE41AAE02@cyrix.home.bogus>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
In-Reply-To: <20061008231659.19EE41AAE02@cyrix.home.bogus>
X-Home-Page: http://www.parisc-linux.org/
User-Agent: Mutt/1.5.9i
X-Virus-Scanned: by amavisd-new-20030616-p10 (Debian) at lackof.org
X-Spam-Checker-Version: SpamAssassin 2.60-bugs.debian.org_2005_01_02 
	(1.212-2003-09-23-exp) on spohr.debian.org
X-Spam-Level: 
X-Spam-Status: No, hits=-4.4 required=4.0 tests=BAYES_00,HAS_BUG_NUMBER,
	UPPERCASE_25_50 autolearn=no version=2.60-bugs.debian.org_2005_01_02

On Sun, Oct 08, 2006 at 04:16:58PM -0700, Matt Taggart wrote:
> No I mean the message from Yanko Kaneti <yaneti@declera.com> which says,
...

Yanko-san didn't have his facts quite right though I agree his
conclusion is correct:

linux-2.6.18/include/linux/pci_ids.h says:
#define PCI_DEVICE_ID_LSI_53C1510       0x000a
#define PCI_DEVICE_ID_NCR_53C1510       0x0010


>  I have with me a Compaq dl380 with the same lspci output and sym2 from
>  the latest kernel tree still claims the raid. I believe the tweak for
>  PCI_DEVICE_ID_LSI_53C1510 in sym_glue.c should also be applied to
>  PCI_DEVICE_ID_NCR_53C1510 to fix this for good."

I didn't know Compaq used two different 53[cC]510 parts.
Patch below adds the same tweak to the 0x0010 device ID.

James or willy, this look good to you?

thanks,
grant

diff --git a/drivers/scsi/sym53c8xx_2/sym_glue.c b/drivers/scsi/sym53c8xx_2/sym_glue.c
index 52abce4..defca91 100644
--- a/drivers/scsi/sym53c8xx_2/sym_glue.c
+++ b/drivers/scsi/sym53c8xx_2/sym_glue.c
@@ -2093,7 +2093,7 @@ static struct pci_device_id sym2_id_tabl
 	{ PCI_VENDOR_ID_LSI_LOGIC, PCI_DEVICE_ID_NCR_53C875,
 	  PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0UL },
 	{ PCI_VENDOR_ID_LSI_LOGIC, PCI_DEVICE_ID_NCR_53C1510,
-	  PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0UL }, /* new */
+	  PCI_ANY_ID, PCI_ANY_ID, PCI_CLASS_STORAGE_SCSI<<8,  0xffff00, 0UL },
 	{ PCI_VENDOR_ID_LSI_LOGIC, PCI_DEVICE_ID_LSI_53C895A,
 	  PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0UL },
 	{ PCI_VENDOR_ID_LSI_LOGIC, PCI_DEVICE_ID_LSI_53C875A,



